src/app/components/home/card-list/card-list.component.ts

```diff
@@ -0,0 +1,182 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { PokemonEntityService } from '../../../services/pokemons-page/pokemon-entity.service';
+import { map, tap } from 'rxjs/operators';
+import { defaultDialogConfig } from '../../../shared/default-dialog-config';
+import { MatDialog } from '@angular/material/dialog';
+import { SingleCardOverviewComponent } from '../single-card-overview/single-card-overview.component';
+import { MultipleCardOverviewComponent } from '../multiple-card-overview/multiple-card-overview.component';
+import { FavoriteEntityService } from 'src/app/services/favorite-pokemons/favorite-entity.service';
+
+@Component({
+  selector: 'app-card-list',
+  templateUrl: './card-list.component.html',
+  styleUrls: ['./card-list.component.scss'],
+})
+export class CardListComponent implements OnInit, OnDestroy {
+  pokemonsPage$: Observable<PokemonsPage[]>;
+  isComparing: boolean = false;
+  pokemonBeforeComparing: Pokemon;
+  pokemonApiOffset: string = '20';
+  queryParams: string = '';
+
+  favoriteSubscription: Subscription;
+  fetchedPokemonsSubscription: Subscription;
+  dialogSubscription: Subscription;
+
+  constructor(
+    private pokemonsEntityService: PokemonEntityService,
+    private dialog: MatDialog,
+    private fetchedPokemonsEntityService: FetchedPokemonsEntityService,
+    private favoriteEntityService: FavoriteEntityService,
+    private loader: NgxUiLoaderService
+  ) {}
+
+  ngOnInit(): void {
+    this.loadStoredPokemons();
+
+    this.pokemonsEntityService.entities$.pipe(
+      tap((item) => {
+        this.pokemonApiOffset = item.length + '';
+      })
+    );
+  }
+
+  ngOnDestroy() {
+    this.tryToUnsuscribe(this.favoriteSubscription);
+    this.tryToUnsuscribe(this.fetchedPokemonsSubscription);
+    this.tryToUnsuscribe(this.dialogSubscription);
+  }
+
+  tryToUnsuscribe(variable: Subscription) {
+    if (variable !== undefined) {
+      variable.unsubscribe();
+    }
+  }
+
+  loadStoredPokemons() {
+    this.pokemonsPage$ = this.pokemonsEntityService.entities$.pipe(
+      map((pokemons) => pokemons)
+    );
+  }
+
+  filterPokemons(data: SearchBarEventArgs) {
+    this.queryParams = data.newValue;
+    if (this.queryParams !== '' && this.queryParams !== undefined) {
+      this.pokemonsPage$ = this.pokemonsEntityService.entities$.pipe(
+        map((pokemons) =>
+          pokemons.filter((item) => item.name.includes(this.queryParams))
+        )
+      );
+    } else {
+      this.loadStoredPokemons();
+    }
+  }
+
+  addToFavorites(pokemon: PokemonsPage) {
+    let favoritesLength = undefined;
+    let removingFavorite = false;
+
+    this.favoriteSubscription = this.favoriteEntityService.entities$.subscribe(
+      (entities) => {
+        favoritesLength = entities.length;
+
+        if (entities.some((item) => item.name == pokemon.name)) {
+          removingFavorite = true;
+        }
+      }
+    );
+    if (removingFavorite) {
```

Comment:
the subscription is gonna run after the function is done, then, if you want to see the value of removingFavorite before that, you must have it inside the subscribe